File: app/src/main/java/faith/changliu/androiddemo/adapters/GamesAdapter.java
package faith.changliu.androiddemo.adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

import faith.changliu.androiddemo.R;
import faith.changliu.androiddemo.data.entities.Game;

public class GamesAdapter extends RecyclerView.Adapter<GamesAdapter.VH> {

	private ArrayList<Game> mGames;

	public GamesAdapter(ArrayList<Game> mGames) {
		this.mGames = mGames;
	}

	@NonNull
	@Override
	public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
		View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.cell_game, parent, false);
		return new VH(v);
	}

	@Override
	public void onBindViewHolder(@NonNull VH holder, int position) {
		holder.bind(mGames.get(position));
	}

	@Override
	public int getItemCount() {
		return mGames.size();
	}

	class VH extends RecyclerView.ViewHolder {

		// views
		private TextView mTvGame;
		private TextView mTvPayoff;
		private TextView mTvStatus;

		private VH(View itemView) {
			super(itemView);

			mTvGame = itemView.findViewById(R.id.mTvGame);
			mTvPayoff = itemView.findViewById(R.id.mTvPayoff);
			mTvStatus = itemView.findViewById(R.id.mTvStatus);
		}

		private void bind(Game game) {
			mTvGame.setText(game.getGame());
			mTvPayoff.setText("" + game.getPayoff());
			mTvStatus.setText(game.getStatus());
		}
	}

}

File: app/src/main/java/faith/changliu/androiddemo/adapters/GamesAdapter.kt
package faith.changliu.androiddemo.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import faith.changliu.androiddemo.R
import faith.changliu.androiddemo.data.entities.Game
import kotlinx.android.synthetic.main.cell_game.view.*

class GamesAdapter(private val games: ArrayList<Game>) : RecyclerView.Adapter<GamesAdapter.ViewHolder>() {
	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
		val view = LayoutInflater.from(parent.context).inflate(R.layout.cell_game, parent, false)
		return ViewHolder(view)
	}

	override fun getItemCount(): Int = games.size

	override fun onBindViewHolder(holder: ViewHolder, position: Int) {
		holder.bind(games[position])
	}

	inner class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
		fun bind(game: Game) {
			itemView.apply {
				mTvGame.text = game.game
				mTvPayoff.text = game.payoff.toString()
				mTvStatus.text = game.status
			}
		}
	}
}